docs: better contribution docs

Standing-Man opened the pull request:
I organized the development setup documentation for external contributors.

- Closes ##9582.

### Key Changes

- Corrected the term for the LFX Mentorship.
- Introduces instructions for setting up the development environment using Dev Containers.
- Fix Windows Subsystem for Linux link
- Moved the section that verifies the installation of make dependencies after the installation instructions, making it easier for contributors to follow.

sxd commented:
Hello @Standing-Man 

You're missing two things here, an issue describing the problem and a better commit message describing what was done and why.

Regards,

Standing-Man committed:
docs: improve documentation for setting up development environment

- Moved the section that verifies the installation of make dependencies after the installation instructions, making it easier for contributors to follow.

- Corrected the term for the LFX Mentorship.

Signed-off-by: StandingMan <jmtangcs@gmail.com>

Standing-Man committed:
Merge branch 'main' into contribution-docs

Standing-Man committed:
docs: add development environment setup using Dev Container

- Introduces instructions for setting up the development environment using Dev Containers.
- fix Windows Subsystem for Linux link

Signed-off-by: StandingMan <jmtangcs@gmail.com>

Standing-Man committed:
Merge branch 'main' into contribution-docs

Standing-Man commented:
Hi @mnencia and @sxd, ready to review.
